### PR TITLE
Initial java-bambou implementation

### PR DESCRIPTION
**Pull request by martin-leclerc**
Initial implementation of Bambou for Java.

### CODE REVIEW

**Comment by dsamaey** [comment]
Nice job Martin.  Can you please add some unit tests too ?  Good code coverage is important.

**Comment by martin-leclerc** [comment]
Thanks! Yeah, I did create a bunch of unit tests. I probably forgot to
submit them. I will when I get back from my vacation.

.. From the Venetian hotel in Vegas :-)

Martin.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.alu.nuage</groupId>
+	<artifactId>bambou</artifactId>
+	<version>1.0.0-SNAPSHOT</version>
+
+	<build>
+		<plugins>
+			<plugin>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.1</version>
```
Comment (dsamaey) [inline pom.xml]: Why not 3.5.1 ?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+				</configuration>
+			</plugin>
+		</plugins>
+	</build>
+
+	<dependencies>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.6.0-rc1</version>
```
Comment (dsamaey) [inline pom.xml]: Why not 2.7.2 ?

**Comment by primalmotion** [comment]
What happens in Vegas stays in Vegas, but don't forget to bring your unit tests back :)

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dsamaey) [inline pom.xml]: XML formatting (indentation)

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		</dependency>
+		<dependency>
+			<groupId>commons-codec</groupId>
+			<artifactId>commons-codec</artifactId>
+			<version>1.9</version>
+		</dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+            <version>1.7.12</version>
```
Comment (dsamaey) [inline pom.xml]: Why not 1.7.18 ?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
+			<version>4.1.6.RELEASE</version>
+		</dependency>
+		<dependency>
+			<groupId>commons-codec</groupId>
+			<artifactId>commons-codec</artifactId>
+			<version>1.9</version>
```
Comment (dsamaey) [inline pom.xml]: Why not 1.10 ?

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	<dependencies>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.6.0-rc1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
+			<version>4.1.6.RELEASE</version>
```
Comment (dsamaey) [inline pom.xml]: Why not 4.2.5.RELEASE ?

File: pom.xml (hunk 1 above)
Comment (martin-leclerc) [inline pom.xml]: Done. Change available in next commit.

File: pom.xml (hunk 2 above)
Comment (martin-leclerc) [inline pom.xml]: Done. Change available in next commit.

File: pom.xml (hunk 4 above)
Comment (martin-leclerc) [inline pom.xml]: Done. Change available in next commit.

File: pom.xml (hunk 5 above)
Comment (martin-leclerc) [inline pom.xml]: Done. Change available in next commit.

File: pom.xml (hunk 6 above)
Comment (martin-leclerc) [inline pom.xml]: Done. Change available in next commit.

File: pom.xml (hunk 3 above)
Comment (martin-leclerc) [inline pom.xml]: Not sure I understand this one. I triggered a Source->Format in Eclipse just to be sure. However, there does not appear to have been any changes with respect to indentation.

**Comment by primalmotion** [comment]
awesome job, thanks!

**Pull request merged**